src/ggrc/maintenance_views/maintenance.py

```
@@ -0,0 +1,151 @@
+          MigrationLog.id.desc()).first()
+      if row and row.is_migration_complete:
+        context['migration_status'] = 'Complete'
+      elif row:
+        context['migration_status'] = 'In progress'
+    except sqlalchemy.exc.ProgrammingError as e:
+      if not re.search(r"""\(1146, "Table '.+' doesn't exist"\)$""",
+                       e.message):
+        raise
+
+  return render_template("maintenance/trigger.html", **context)
+
+
+def run_migration():
+  """Triggers a deferred task for migration."""
+  try:
+    sess = db.session
+    maint_row = sess.query(Maintenance).get(1)
+    mig_row = MigrationLog(is_migration_complete=False)
+    sess.add(mig_row)
+
+    # Set the db flag before running migrations
+    if maint_row:
+      maint_row.under_maintenance = True
+    else:
+      maint_row = Maintenance(under_maintenance=True)
+      sess.add(maint_row)
+    sess.commit()
+  except sqlalchemy.exc.ProgrammingError as e:
+    if re.search(r"""\(1146, "Table '.+' doesn't exist"\)$""", e.message):
+      mig_row = None
+    else:
+      raise
+
+  mig_row_id = mig_row.id if mig_row else 0
+  deferred.defer(migrate.migrate, row_id=mig_row_id, _queue='ggrc')
+  session['migration_started'] = True
+  return mig_row_id
+
+
+@maintenance_app.route('/maintenance/migrate', methods=['POST'])
+def authenticate():
+  """Authenticates user and allows to run migration."""
+  if "access_token" in request.form:
+    if hasattr(settings, 'ACCESS_TOKEN'):
+      if request.form.get("access_token") == settings.ACCESS_TOKEN:
+        mig_row_id = run_migration()
+        return "{0},{1},{2}".format('Migration is running in background.',
+                                    200,
+                                    mig_row_id)
+      else:
+        msg = "Invalid access token"
+        logger.info(msg)
+        return msg
+    else:
+      msg = "Access token not found in settings"
+      logger.info(msg)
+      return msg
```

Again this shouldn't return 2xx, it should return a server error 5xx

---

Done